File: app/src/main/java/com/kiquenet/introduceme/ui/di/scope/PerActivity.java
package com.kiquenet.introduceme.ui.di.scope;

/**
 * A scoping annotation to permit objects whose lifetime should
 * conform to the life of the Activity to be memorised in the
 * correct component.
 * @author n.diazgranados
 */

import javax.inject.Scope;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Scope
@Retention(RetentionPolicy.RUNTIME)
public @interface PerActivity {
}

File: app/src/main/java/com/kiquenet/introduceme/di/scope/PerActivity.kt
package com.kiquenet.introduceme.di.scope

/**
 * A scoping annotation to permit objects whose lifetime should
 * conform to the life of the Activity to be memorised in the
 * correct component.
 * @author n.diazgranados
 */

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import javax.inject.Scope

@Scope
@Retention(RetentionPolicy.RUNTIME)
annotation class PerActivity